### PR TITLE
Fix helm valuesFrom escape dot notation

### PR DESCRIPTION
**Pull request by Lingkar**
#870 Should fix the helm ValuesFrom escaping of dot notation.

### CODE REVIEW

**Review by allenporter — CHANGES_REQUESTED** [review]
Thank you for the report and fix. Reading the spec, are there other cases we need to support/escape too like the other characters mentioned? I saw also there is some kind of list support but that can come later

**Comment by Lingkar** [comment]
@allenporter yeah I was also thinking about this, but as the code only splits on the '.' I initially didn't think we needed to.
However, we actually might need to remove the non-escaped back slashes in front of the other special characters.

**Comment by allenporter** [comment]
Looks like the full logic for this is here: https://github.com/helm/helm/blob/31e22b9866af91e1a0ea2ad381798f6c5eec7f4f/pkg/strvals/parser.go#L179

File: flux_local/values.py [hunk 1]
```diff
@@ -198,7 +198,23 @@ def _update_helmrelease_values(
     """Expand value references in the HelmRelease."""
 
     if ref.target_path:
-        parts = ref.target_path.split(".")
+        raw_parts = ref.target_path.split(".")
+
+        # Merge escaped parts
+        parts = []
+        i = 0
+        while i < len(raw_parts):
```
[the remark this answers is not in the document]
Comment (Lingkar) [reply flux_local/values.py:206]: I don't believe so.
From the implemented test, we want the following:
ingress.annotations.azure\.workload\.identity/client-id

to become:
```
ingress:
  annotations: 
    azure.workload.identity/client-id: "SECRET_VALUE"
```

By replacing the escaped dots for normal dots and than splitting up the parts we are creating the following structure:

```
ingress:
  annotations: 
    azure:
      workload:
        identity/client-id: "SECRET_VALUE"
```

And consequently running into the same Go unmarshal error:
`Error: YAML parse error on podinfo/templates/ingress.yaml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal object into Go struct field .metadata.annotations of type string`

**Comment by Lingkar** [comment]
> Looks like the full logic for this is here: https://github.com/fluxcd/pkg/blob/e0f6da054dd76a154d604ae2f01c6dd30c6ef80c/chartutil/values.go#L257 and https://github.com/helm/helm/blob/31e22b9866af91e1a0ea2ad381798f6c5eec7f4f/pkg/strvals/parser.go#L179

@allenporter I will take a look and add the other characters as well.

File: flux_local/values.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (allenporter) [reply flux_local/values.py:206]: Got it! I didn't understand what the point of this was, now I do and this example is very helpful!

(1) Lets add specific unit tests in tests/test_values.py to exercise corner cases in addition to the end to end test
(2) Lets try to use a regexp for this. I realize regexps add new complexity, but it think its worth it to reduce the lines of code here. I think we can split this into two steps.

First (a) do the splitting by `.` except when escaped we can use the regexp:

```shell
$ python
>>> import re
>>> s = "ingress.annotations.azure\\.workload\\.identity/client-id"
# shows the bug
>>> s.split(".")
['ingress', 'annotations', 'azure\\', 'workload\\', 'identity/client-id']
# Split on a "." except for escaped "\\"
>>> re.split(r"[^\\]\.",s)
['ingres', 'annotation', 'azure\\.workload\\.identity/client-id']

Second (b) then its ok to loop through and unescape
```shell
>>> raw_parts = re.split(r"[^\\]\.",s)
>>> raw_parts
['ingres', 'annotation', 'azure\\.workload\\.identity/client-id']
>>> parts = [raw_part.replace("\\.", ".") for raw_part in raw_parts ]
>>> parts
['ingres', 'annotation', 'azure.workload.identity/client-id']
```

That replace function couuld be extracted into a helper function, but just there as an example.

With that approach we can take advantage of python syntax to shrink it down quite a bit.

**Comment by allenporter** [comment]
Thank you @Lingkar appreciate the contribution!

**Comment by allenporter** [comment]
Feel free to add the other characters in another PR but agree that fixing the dot seems like the priority so I went ahead and merged.
